controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/resource/Hosts.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pinterest.rocksplicator.controller.resource;
+
+import org.hibernate.validator.constraints.NotEmpty;
+
+import java.util.List;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * @author Ang Xu (angxu@pinterest.com)
+ */
+@Path("/v1/clusters/{clusterName : [a-zA-Z0-9\\-_]+}/hosts")
+public class Hosts {
+
+  /**
+   * Adds a list of hosts to a given cluster to make it complete.
+   *
+   * @param clusterName name of the cluster
+   * @param hosts       a list of hosts to add, in the format of "ip:port"
+   */
+  @POST
+  @Path("/addHosts")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public void addHosts(@PathParam("clusterName") String clusterName,
```

Comment:
expanding/shrinking will be done in "resize" cluster.
this is similar to "add_without_move_shards" in realpin which add back missing hosts.